src/readsums.c

```diff
@@ -145,6 +152,14 @@ static rs_result rs_loadsig_s_stronglen(rs_job_t *job)
 
     job->statefn = rs_loadsig_s_weak;
     
+    /* Estimate the number of blocks stored in signature */
+    /* Magic+header is 12 bytes,
+       each block thereafter is 4 bytes weak_sum+strong_sum_len bytes */
+    if (job->sig_size > 0) {
+        job->signature->count = (job->sig_size-12)/(4+job->strong_sum_len);
```

Comment:
I think rather than putting this in `signature->count` I would have a separate `job->estimated_signature_count`, and keep the signature object for just what was actually read.